src/iface/ethernet.rs

```diff
@@ -761,11 +764,63 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
             // Ignore any echo replies.
             Icmpv6Repr::EchoReply { .. } => Ok(Packet::None),
 
+            // Forward any NDISC packets to the ndisc packet handler
+            Icmpv6Repr::Ndisc(repr) if ip_repr.hop_limit() == 0xff => match ip_repr {
+                IpRepr::Ipv6(ipv6_repr) => self.process_ndisc(timestamp, ipv6_repr, repr),
+                _ => Ok(Packet::None)
+            },
+
             // FIXME: do something correct here?
             _ => Err(Error::Unrecognized),
         }
     }
 
+    #[cfg(feature = "proto-ipv6")]
+    fn process_ndisc<'frame>(&mut self, timestamp: Instant, ip_repr: Ipv6Repr,
+                             repr: NdiscRepr<'frame>) -> Result<Packet<'frame>> {
+        let packet = match repr {
+            // TODO: Use the neighbor advert flags and handle the advertisement acordingly
```

Comment:
What exactly is not being done appropriately here?

Reply:
We don't consider the Override flag. Essentially we assume that Override is always set.

Reply:
What are the implications of this?

Reply:
Honestly shouldn't be too hard to implement. I'll just implement support for the override flag.